jobs/delorean/jenkinsfiles/branch/suite-next/Jenkinsfile

```diff
@@ -0,0 +1,98 @@
+#!groovy
+
+// Ensure required parameters are provided
+def verifyParameters(suiteParams) {
+  if (!suiteParams.gitUrl || !suiteParams.suiteBranch) {
+      def userInput = input message: 'Suites Next Options', parameters: [
+        string(defaultValue: (suiteParams.gitUrl ?: 'git@github.com:integr8ly/installation.git'), description: 'The installation repo git url', name: 'installationGitUrl'),
+        string(defaultValue: (suiteParams.suiteBranch ?: ''), description: 'Desired name for the suite branch', name: 'suiteBranch'),
+      ]
+      suiteParams.gitUrl = userInput.installationGitUrl
+      suiteParams.suiteBranch = userInput.suiteBranch
+      verifyParameters(suiteParams)
+  }
+}
+
+// Checkout remote repository
+def checkoutGitRepo(gitUrl, gitRef, credentialsID) {
+  checkout([$class: 'GitSCM', branches: [[name: gitRef]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: credentialsID, url: gitUrl]]])
+}
+
+// Get product next branches available in the installation repo
+def getProductNextBranches(gitOrg, gitRepo, sourceBranch) {
+  nextBranches = sh(returnStdout: true, script: "git ls-remote origin | grep next | awk '{print \$2}'").replaceAll("refs/heads/", "").split()
+
+  // Do not include the source branch in the next branches list
+  productNextBranches = nextBranches.minus(sourceBranch)
+  return productNextBranches
+}
+
+def gitCreateAndCheckoutBranch(branchName, pushOnCreate = false) {
+  String remoteBranchCommit = sh(returnStdout: true, script: "git ls-remote origin refs/heads/${branchName} | cut -f 1").trim()
+  if (remoteBranchCommit) {
+    sh "git push origin --delete ${branchName}"
+  }
```

Comment:
I'd probably remove this from here and make it an explicit step in an appropriate stage. It would also be worthwhile adding some guard around this to make sure we don't accidentally delete the wrong branch.

You could also just not have an explicit delete step and force push at the end. That way you are always just creating a new branch from master and replacing any that might already exist. This would still require a check to make sure we don;t do it to any branches other than these suite ones.